packages/pluggableWidgets/line-chart-native/src/utils/SeriesDataLoadingFunction.ts

```diff
@@ -0,0 +1,203 @@
+
+    if (!dataSourceItemGroups) {
+        return null;
+    }
+
+    const loadedSeries: LineChartSeries[] = [];
+
+    for (const itemGroup of dataSourceItemGroups) {
+        const dataPoints = extractDataPoints(series, itemGroup.items);
+
+        if (!dataPoints) {
+            return null;
+        }
+
+        loadedSeries.push({
+            dataPoints,
+            interpolation,
+            lineStyle: dynamicLineStyle,
+            name: itemGroup.dynamicSeriesNameValue,
+            stylePropertyName: itemGroup.dynamicStylePropertyNameValue
+        });
+    }
+
+    return loadedSeries;
+}
+
+function groupDataSourceItems(series: SeriesType): DataSourceItemGroup[] | null {
+    const { dynamicDataSource, dynamicSeriesName, dynamicStylePropertyName, groupByAttribute, type } = series;
+
+    if (type !== "dynamic") {
+        throw Error("Expected series to be dynamic");
+    }
+
+    const dataSource = ensure(dynamicDataSource);
+
+    if (!dataSource.items) {
+        return null;
+    }
+
+    const dataSourceItemGroupsResult: DataSourceItemGroup[] = [];
+
+    for (const item of dataSource.items) {
+        const groupByAttributeValue = ensure(groupByAttribute)(item);
+
+        if (
+            !groupByAttributeValue.value &&
+            typeof groupByAttributeValue.value !== "string" &&
+            typeof groupByAttributeValue.value !== "boolean"
+        ) {
+            return null;
+        }
+
+        const group = dataSourceItemGroupsResult.find(group => {
+            if (groupByAttributeValue.value instanceof Date && group.groupByAttributeValue instanceof Date) {
+                return group.groupByAttributeValue.getTime() === groupByAttributeValue.value.getTime();
+            } else if (groupByAttributeValue.value instanceof Big && group.groupByAttributeValue instanceof Big) {
+                return group.groupByAttributeValue.eq(groupByAttributeValue.value);
+            }
+            return group.groupByAttributeValue === groupByAttributeValue.value;
+        });
+
+        if (group) {
+            group.items.push(item);
+        } else {
+            const newDataSourceItemGroup: DataSourceItemGroup = {
+                groupByAttributeValue: groupByAttributeValue.value,
+                items: [item]
+            };
+
+            if (dynamicSeriesName) {
+                const dynamicSeriesNameValue = dynamicSeriesName(item);
+
+                if (!dynamicSeriesNameValue.value && typeof dynamicSeriesNameValue.value !== "string") {
+                    return null;
+                }
+
+                newDataSourceItemGroup.dynamicSeriesNameValue = dynamicSeriesNameValue.value;
+            }
+
+            if (dynamicStylePropertyName) {
+                const dynamicStylePropertyNameValue = dynamicStylePropertyName(item);
+
+                if (!dynamicStylePropertyNameValue.value && typeof dynamicStylePropertyNameValue.value !== "string") {
```

Comment:
Same question about the second check here

Reply:
See other comment. Fixed it.